modules/core/src/main/java/org/apache/synapse/config/xml/endpoints/EndpointDefinitionFactory.java

```diff
@@ -159,20 +161,15 @@ public EndpointDefinition createDefinition(OMElement elem) {
                 String d = duration.getText();
                 if (d != null) {
                     try {
-                        Pattern pattern = Pattern.compile("\\{.*\\}");
-                        if (pattern.matcher(d).matches()) {
-                            d = d.trim().substring(1, d.length() - 1);
-                            SynapseXPath xpath = new SynapseXPath(d);
-                            definition.setDynamicTimeoutExpression(xpath);
+                        if (isExpression(d)) {
```

Comment:
Do we need this check? The ValueFactory will check wether it's an expression or not and resolve the value.

Reply:
Updated using ValueFactory.